docker/local/polygon-edge.sh

```diff
@@ -8,51 +8,125 @@ CHAIN_CUSTOM_OPTIONS=$(tr "\n" " " << EOL
-                --bootnode "/dns4/node-2/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[1] | .node_id')"
-         fi
+          "ibft")
+              if [ -f "$GENESIS_PATH" ]; then
+                  echo "Secrets have already been generated."
+              else
+                  echo "Generating iBFT secrets..."
+                  secrets=$("$POLYGON_EDGE_BIN" secrets init --insecure --num 4 --data-dir /data/data- --json)
+                  echo "Secrets have been successfully generated"
+
+                  rm -f /data/genesis.json
+
+                  echo "Generating iBFT Genesis file..."
+                  "$POLYGON_EDGE_BIN" genesis $CHAIN_CUSTOM_OPTIONS \
+                    --dir /data/genesis.json \
+                    --consensus ibft \
+                    --ibft-validators-prefix-path data- \
+                    --bootnode "/dns4/node-1/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[0] | .node_id')" \
+                    --bootnode "/dns4/node-2/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[1] | .node_id')" \
+                    --bootnode "/dns4/node-3/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[2] | .node_id')" \
+                    --bootnode "/dns4/node-4/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[3] | .node_id')"
+              fi
               ;;
           "polybft")
               echo "Generating PolyBFT secrets..."
               secrets=$("$POLYGON_EDGE_BIN" polybft-secrets init --insecure --num 4 --data-dir /data/data- --json)
               echo "Secrets have been successfully generated"
 
-              echo "Generating PolyBFT Genesis file..."
+              if [ -z "$ROOTCHAIN_IP" ]
+              then
+                    echo "\$ROOTCHAIN_IP evar must be provided with rootchain's IP address"
+                    exit 1
+              fi
+
+              rm -f /data/genesis.json
+
+              echo "Generating PolyBFT genesis file..."
               "$POLYGON_EDGE_BIN" genesis $CHAIN_CUSTOM_OPTIONS \
                 --dir /data/genesis.json \
                 --consensus polybft \
-                --validators-path /data \ 
+                --validators-path /data \
                 --validators-prefix data- \
+                --reward-wallet 0xDEADBEEF:1000000 \
                 --bootnode "/dns4/node-1/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[0] | .node_id')" \
-                --bootnode "/dns4/node-2/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[1] | .node_id')"
+                --bootnode "/dns4/node-2/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[1] | .node_id')" \
+                --bootnode "/dns4/node-3/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[2] | .node_id')" \
+                --bootnode "/dns4/node-4/tcp/1478/p2p/$(echo "$secrets" | jq -r '.[3] | .node_id')"
+
+              echo "Deploying stake manager..."
+              "$POLYGON_EDGE_BIN" polybft stake-manager-deploy \
+                --jsonrpc http://$ROOTCHAIN_IP:8545 \
```

Comment:
I was running the following command:
```bash
stefan@Stefan:~/go/src/polygon-edge/docker/local$ docker-compose build && docker-compose up -d
```
...and encountered an error here:
`faield to send local transaction: dialing to the given TCP address timed out`

Is it possible that provided rootchain IP address is not correct?

P.s. the same result is when running 
```bash
scripts/cluster polybft --docker
```

Reply:
Yeah, the rootchain address is not correct but it works on my local env. Did you use the latest version of it?

Reply:
Indeed it was the latest version… Not sure, worth that somebody else tries out.

Reply:
Try now

Reply:
Working fine now.